src/utils.cpp

```diff
@@ -764,5 +771,250 @@ bool executePreAction(const nlohmann::json& i_parsedConfigJson,
+                std::get_if<types::IpzType>(&i_paramsToReadData))
+        {
+            l_interface = constants::ipzVpdInf + std::get<0>(*l_ipzData);
+            l_keyword = std::get<1>(*l_ipzData);
+        }
+
+        return (
+            utils::readDbusProperty(l_service, i_path, l_interface, l_keyword));
+    }
+}
+
+std::string getInvPath(const nlohmann::json i_jsonObj,
+                       const std::string& i_path)
+{
+    // Given any path, fetch the inventory path from json
+    if (i_path.empty())
+    {
+        throw(DataException("Invalid File path provided."));
+    }
+
+    if (!i_jsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in VPD JSON");
+    }
+
+    const nlohmann::json& listOfFrus =
+        i_jsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& itemFRUS : listOfFrus.items())
+    {
+        const auto l_hwPath = itemFRUS.key();
+        const auto l_redundantHwPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("redundantEeprom", "");
+        const auto l_invPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("inventoryPath", "");
+
+        if (i_path == l_hwPath || i_path == l_redundantHwPath ||
+            i_path == l_invPath)
+        {
+            return l_invPath;
+        }
+    }
+
+    return std::string();
+}
+
+std::string getRedundantPath(const nlohmann::json i_jsonObj,
+                             const std::string& i_path)
+{
+    // Given any path, fetch the redundant eeprom path from json
+    if (i_path.empty())
+    {
+        throw(DataException("Invalid File path provided."));
+    }
+
+    if (!i_jsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in VPD JSON");
+    }
+
+    const nlohmann::json& listOfFrus =
+        i_jsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& itemFRUS : listOfFrus.items())
+    {
+        const auto l_hwPath = itemFRUS.key();
+        const auto l_redundantHwPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("redundantEeprom", "");
+        const auto l_invPath =
+            i_jsonObj["frus"][l_hwPath].at(0).value("inventoryPath", "");
+
+        if (i_path == l_hwPath || i_path == l_redundantHwPath ||
+            i_path == l_invPath)
+        {
+            return l_redundantHwPath;
+        }
+    }
+
+    return std::string();
+}
+
+std::string getPrimaryPath(const nlohmann::json i_jsonObj,
+                           const std::string& i_path)
+{
+    // Given any path, fetch the primary hardware path from json
+    if (i_path.empty())
+    {
+        throw std::runtime_error("Invalid File path provided.");
+    }
+
+    if (!i_jsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in VPD JSON");
+    }
+
+    if (i_jsonObj["frus"].contains(i_path))
+    {
+        return i_path;
+    }
+
+    const nlohmann::json& listOfFrus =
```

Comment:
local variable name should starts with l_* for listOfFrus & itemFRUS

Reply:
done